src/Trace/Trace.php

```diff
@@ -46,29 +45,27 @@ class Trace implements \Serializable
     private $traceId;
 
     /**
-     * @var TraceSpan[] List of TraceSpans to report
+     * @var Span[] List of Span to report
      */
     private $spans = [];
 
     /**
      * Instantiate a new Trace instance.
      *
-     * @param ConnectionInterface $connection The connection to Stackdriver Trace.
      * @param string $projectId The id of the project this trace belongs to.
      * @param string $traceId [optional] The id of the trace. If not provided, one will be generated
      *        automatically for you.
-     * @param array $spans [optional] Array of TraceSpan constructor arguments. See
-     *        {@see Google\Cloud\Trace\TraceSpan::__construct()} for configuration details.
+     * @param array $spans [optional] Array of Span constructor arguments. See
+     *        {@see Google\Cloud\Trace\Span::__construct()} for configuration details.
      * }
      */
-    public function __construct(ConnectionInterface $connection, $projectId, $traceId = null, $spans = null)
+    public function __construct($projectId, $traceId = null, $spans = null)
     {
-        $this->connection = $connection;
         $this->projectId = $projectId;
         $this->traceId = $traceId ?: $this->generateTraceId();
         if ($spans) {
-            $this->spans = array_map(function ($span) {
-                return new TraceSpan($span);
+            $this->spans = array_map(function ($span) use ($projectId, $traceId) {
+                return new Span($traceId, $span);
```

Comment:
This should be `return new Span($traceId, ['spanId' => $span]);`, unless I'm missing something. Unless I'm incorrect, this also suggests a gap in test coverage, considering Travis is green at the moment. :)

Reply:
Here, `$span` is an array of data. I'll rename to `$spanData`.

Reply:
Cool! I figured I might have been missing something. sorry for the noise.